bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -0,0 +1,572 @@
+                            currentVolumeHash = 0L;
+                        } catch (VizioException e) {
+                            logger.debug("Unable to set volume on the Vizio TV, command volume: {}, Exception: {}",
+                                    command, e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        } catch (NumberFormatException e) {
+                            logger.debug("Unable to parse command volume value {} as int", command);
+                        }
+                    }
+                    break;
+                case MUTE:
+                    debounce = false;
+                    synchronized (sequenceLock) {
+                        try {
+                            if (command == OnOffType.ON && !currentMute) {
+                                communicator.sendKeyPress(KeyCommand.MUTETOGGLE.getJson());
+                                currentMute = true;
+                            } else if (command == OnOffType.OFF && currentMute) {
+                                communicator.sendKeyPress(KeyCommand.MUTETOGGLE.getJson());
+                                currentMute = false;
+                            }
+                        } catch (VizioException e) {
+                            logger.debug("Unable to send mute {} command to the Vizio TV, Exception: {}", command,
+                                    e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        }
+                    }
+                    break;
+                case SOURCE:
+                    debounce = false;
+                    synchronized (sequenceLock) {
+                        try {
+                            // if input changed again before polling has run, get current input hash from the TV
+                            // first
+                            if (currentInputHash.equals(0L)) {
+                                CurrentInput polledInput = communicator.getCurrentInput();
+                                if (!polledInput.getItems().isEmpty()) {
+                                    currentInputHash = polledInput.getItems().get(0).getHashval();
+                                }
+                            }
+                            communicator
+                                    .changeInput(String.format(MODIFY_STRING_SETTING_JSON, command, currentInputHash));
+                            currentInputHash = 0L;
+                        } catch (VizioException e) {
+                            logger.debug("Unable to set current source on the Vizio TV, source: {}, Exception: {}",
+                                    command, e.getMessage());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
Consider if the message above or even the exception message should be included in this status update to provide more immediate information to the user.